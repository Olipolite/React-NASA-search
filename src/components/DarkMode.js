import React, { useEffect, useState } from "react";
import "../styles/darkmode.css";

const DarkMode = () => {
    const [darkMode, setDarkMode] = useState('light')
    const toggleTheme = () => {
        if ( darkMode === 'light') {
            setDarkMode('dark')
        } else {
            setDarkMode('light')
        }
    };
    useEffect(() => {
        document.body.className = darkMode;
    }, [darkMode]);
    return (
        <div className={`App ${darkMode}`}>
            <button onClick={toggleTheme}>Toggle Theme</button>
        </div>
    );
}
export default DarkMode;
