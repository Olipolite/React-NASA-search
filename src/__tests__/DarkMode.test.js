import { fireEvent, render, screen } from '@testing-library/react';
import DarkMode from '../components/DarkMode';

describe("DarkMode", () => {

    it("renders correctly", () => {
      const { asFragment } = render(<DarkMode />);

      expect(asFragment()).toMatchSnapshot();
    });

    it("default state works properly", () => {
      render(<DarkMode />)
      const div = screen.getByTestId('div');
      
      expect(div).toHaveClass("light");
    });

    it("button toggles Darkmode when clicked", () => {
      render(<DarkMode />)

      const div = screen.getByTestId('div');
      const button = screen.getByRole('button');
      fireEvent.click(button);

      expect(div).toHaveClass("dark");

      fireEvent.click(button);

      expect(div).toHaveClass("light");
    })
  });
