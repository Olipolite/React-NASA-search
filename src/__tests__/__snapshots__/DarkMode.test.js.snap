// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DarkMode renders correctly 1`] = `
<DocumentFragment>
  <div
    class="light"
    data-testid="div"
  >
    <button
      class="dark-mode-button"
    >
      Dark Mode
    </button>
  </div>
</DocumentFragment>
`;
